docs/proposals/multiple-sources-for-applications.md

```diff
@@ -0,0 +1,225 @@
+
+### Non-goals
+* Allow reconciliation from Argo CD Application resources that are located at various sources. We believe this would be possible with some adaptions in the reconcilation workflow.
+
+## Proposal
+
+### Add new `sources` field in Application Spec
+
+The proposal is to add a new field `sources` which would allow users to input list of `ApplicationSource`s. We would mark field `source` as deprecated and would override the details under `source` with details under `sources` field.
+
+```yaml
+spec:
+  source:
+    repoURL: https://charts.bitnami.com/bitnami
+    targetRevision: 8.5.8
+    helm:
+    valueFiles:
+        - values.yaml
+    chart: mysql
+  sources:                                          # new field
+    - repoURL: https://charts.bitnami.com/bitnami
+      targetRevision: 8.5.8
+      helm:
+        valueFiles:
+          - values.yaml
+      chart: mysql
+```
+
+### Add `externalValues` field in helm section of Application Spec
+
+Along with new `sources` field, add a new field for accepting external value files `externalValueFiles` under the `helm` section of each ApplicationSource.
+
+```yaml
+sources:
+    - repoURL: https://charts.bitnami.com/bitnami
+      targetRevision: 8.5.8
+      helm:
+        valueFiles:
+          - values.yaml
+        externalValueFiles:                         # new field
+          - repoURL: https://github.com/KaiReichart/argo-test-values.git
+            targetRevision: main
+            valueFiles:
+              - values.yaml
+      chart: mysql
+```
+
+### Combined Example Application yaml
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: grafana
+  namespace: argocd
+spec:
+  destination:
+    namespace: monitoring
+    server: https://some.k8s.url.com:6443
+  project: default
+  source:
+    repoURL: https://charts.bitnami.com/bitnami
+    targetRevision: 8.5.8
+    helm:
+    valueFiles:
+        - values.yaml
+    chart: mysql
+  sources:                                          # new field
+    - repoURL: https://charts.bitnami.com/bitnami
+      targetRevision: 8.5.8
+      helm:
+        valueFiles:
+          - values.yaml
+        externalValueFiles:                         # new field
+          - repoURL: https://github.com/KaiReichart/argo-test-values.git
+            targetRevision: main
+            valueFiles:
+              - values.yaml
+      chart: mysql
+    - repoURL: https://charts.bitnami.com/bitnami1
+      targetRevision: 8.5.8
+      helm:
+        valueFiles:
+          - values.yaml
+        externalValueFiles:
+          - repoURL: https://github.com/KaiReichart/argo-test-values.git
+            targetRevision: main
+            valueFiles:
+              - values.yaml
+      chart: mysql
```

Comment:
Can this example be made more real-world (i.e. using meaningfully different charts/repos)?